src/microvis/core/_vis_model.py

```diff
@@ -177,21 +177,22 @@ def _on_any_event(self, info: EmissionInfo) -> None:
 # XXX: also ... this might make more sense as a method on the VisModel class
 # where we have access to the bound "T" type variable (could remove some casts)
 @lru_cache
-def validate_backend_class(
-    cls: type[VisModel], backend_class: Any
+def validate_adaptor_class(
+    model_class: Type[VisModel], adaptor_class: Any
 ) -> type[BackendAdaptor]:
-    """Validate that the backend class is appropriate for the object."""
-    logger.debug(f"Validating backend class {backend_class} for {cls}")
+    """Validate that the adaptor class is appropriate for the core object."""
+    logger.debug(f"Validating adaptor class {adaptor_class} for {model_class}")
     if missing := {
         SETTER_METHOD.format(name=signal._name)
-        for signal in cls.__signal_group__._signals_.values()
-        if not hasattr(backend_class, SETTER_METHOD.format(name=signal._name))
+        for signal in model_class.__signal_group__._signals_.values()
+        if
+        not hasattr(adaptor_class, SETTER_METHOD.format(name=signal._name))
     }:
         raise ValueError(
-            f"{backend_class} cannot be used as a backend object for {cls}: "
-            f"it is missing the following setters: {missing}"
+            f"{adaptor_class} cannot be used as a backend object for "
+            f"{model_class}: it is missing the following setters: {missing}"
         )
-    return cast("Type[BackendAdaptor]", backend_class)
+    return cast("Type[BackendAdaptor]", adaptor_class)
```

Comment:
This logic checks for the presence of `set_` methods for all signals rather than directly checking the adaptor implements all abstract methods of the associated protocol - do you agree this is a little indirect? This is what is causing the validation failures with computed property/setter for size of the canvas

I toyed with moving this onto the class, finding a reference to the protocol and checking the names of the abstract methods but my metaprogramming-fu is not so strong - before I invest much more time, do you think this is the right path forwards?

note: when that method is on the class you can't keep `@lru_cache` without doing something to satisfy pydantic, it starts complaining about not having a validator for lru caches

Reply:
@tlambert03 your comment above definitely implies the method is the preferred path - thanks for the info! What do you think about the proposed change of validation logic above?

Reply:
Awesome, yea all makes sense, using the signal is wrong.  However, I also am hesitant to use the protocol either (though it's technically correct) since it's a manually constructed thing based on the model.  Maybe for now we can use the names of **fields** of the model (rather than the signals)?  And make a note that if we need to relax that to the protocols later it would be fine.

As for the cache, just remove it, I'll fix it with a different approach later

Reply:
went with this - canvas is still proving a little tricky, validation is failing because the setter '_vis_set_views' is unavailable, the protocol has `_vis_add_view`